app/main.py

```diff
@@ -1,8 +1,33 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        self.people[self.name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    people_list = []
+
+    for person in people:
+        name = person["name"]
+        age = person["age"]
+        new_person = Person(name, age)
+        people_list.append(new_person)
+
+        if "wife" in person:
+            wife_name = person["wife"]
+            if wife_name in Person.people:
```

Comment:
It's better to use `and` instead of nested `if` blocks
Also read about dict.get() method, it will be even better in this case